uro98/assignment3/FindAllWords.java

```diff
@@ -0,0 +1,80 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class FindAllWords {
+  
+  /**
+   * Find all the words in the dictionary that can be formed in the grid.
+   * @param grid
+   * @param dictionary
+   * @return
+   */
+  public static Set<String> findAllWords(Grid grid, Dictionary dictionary) {
+    Set<String> totalWords = new HashSet<String>();
+    
+    // For each position in the grid, find all the words that can be formed starting from there
+    for (int i = 0; i < grid.getHeight(); i++) {
+      for (int j = 0; j < grid.getWidth(); j++) {
+        Set<String> wordSet = FindAllWords.searchWords(
+            grid,
+            dictionary,
+            new boolean[grid.getHeight()][grid.getWidth()],
+            new HashSet<String>(),
+            new ArrayList<String>(),
+            i,
+            j);
+        totalWords.addAll(wordSet);
+      }
+    }
+    
+    return totalWords;
+  }
+  
+  public static Set<String> searchWords(
```

Comment:
I think this should be private, since it is an implementation detail. From user's perspective `findAllWords` is the only interface they need.

Also: even though it should be a private method, its interface is quite complicated, so it would be good to add comment to explain how it should be used.